src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -31,6 +33,12 @@ public IEnumerator<TElement> GetEnumerator()
                 {
                     yield return buffer._items[map[i]];
                 }
+
+                var rentedMap = Interlocked.Exchange(ref map, null);
```

Comment:
ARM has a weaker memory model by default so yes. Don't know if there is a better multi-threaded guard though. Unless just guard against double dispose and not multi threading?